windows/wintools_uninstall_verify/prepare_pvscsi_vmxnet3_device.yml

```diff
@@ -0,0 +1,69 @@
+# Copyright 2022 VMware, Inc.
+# SPDX-License-Identifier: BSD-2-Clause
+---
+# Get PVSCSI and VMXNET3 device driver info, if there are no PVSCSI
+# and/or VMXNET3 deivces on VM, add new devices
+#
+- name: "Initialize the facts of VM with PVSCSI and VMXNET3 devices"
+  ansible.builtin.set_fact:
+    vm_has_pvscsi_boot_disk: false
+    vm_has_pvscsi_ctl: false
+    vm_has_vmxnet3_net: false
+
+# Check if boot disk controller is PVSCSI
+- name: "Set fact of VM with PVSCSI boot disk controller"
+  ansible.builtin.set_fact:
+    vm_has_pvscsi_boot_disk: true
+  when:
+    - new_vm is defined and new_vm | bool
+    - boot_disk_controller is defined and boot_disk_controller == 'paravirtual'
+- name: "Get boot disk controller type for existing VM"
+  block:
+    - include_tasks: ../utils/win_get_boot_disk_ctl_type.yml
+    - name: "Set fact of VM with PVSCSI boot disk controller"
+      ansible.builtin.set_fact:
+        vm_has_pvscsi_boot_disk: true
+      when:
+        - win_boot_disk_ctl_type == 'paravirtual'
+  when: new_vm is undefined or not new_vm | bool
+
+# If not PVSCSI boot disk controller
+- name: "Check if VM with PVSCSI controller"
+  block:
+    - include_tasks: ../utils/win_get_pvscsi_ctl_num.yml
+    - name: "Set fact of VM with PVSCSI controller"
+      ansible.builtin.set_fact:
+        vm_has_pvscsi_ctl: true
+      when: ctl_num_guest | int > 0
+  when: not vm_has_pvscsi_boot_disk
+
+- include_tasks: ../utils/win_get_netadapter_num.yml
+- name: "Set fact of VM with VMXNET3 network adapter"
+  ansible.builtin.set_fact:
+    vm_has_vmxnet3_net: true
+  when: win_get_netadapter_num_dict.VMXNET3 | int > 0
+
+# Add a new PVSCSI controller if it's not boot disk controller and
+# VM has no such device
+- name: "Add a new PVSCSI controller to VM"
+  block:
+    - include_tasks: ../../common/vm_hot_add_remove_disk_ctrl.yml
+      vars:
+        disk_controller_ops: "present"
+        disk_controller_type: "paravirtual"
+    - name: "Check add new PVSCSI controller config changes"
+      ansible.builtin.assert:
+        that:
+          - disk_controller_facts is defined
+          - disk_controller_facts.changed
+        fail_msg: "Add new PVSCSI controller task result 'changed' is not true."
+  when:
+    - not vm_has_pvscsi_boot_disk
```

Comment:
No need the first condition if vm_has_pvscsi_ctl is set in above line 16

Reply:
If you struggle with one more line then if set `vm_has_pvscsi_ctl = true` will increase 2 more lines. Just as I said above this paramter set or not set doesn't matter, it's not used later, here just distinguish boot disk controller or non-boot disk controller.